tap/api/tcp_streams_map.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"runtime"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	"github.com/up9inc/mizu/shared/logger"
+	"github.com/up9inc/mizu/tap/api/diagnose"
+)
+
+type TcpStreamMap struct {
+	Streams  *sync.Map
+	streamId int64
+}
+
+func NewTcpStreamMap() *TcpStreamMap {
+	return &TcpStreamMap{
+		Streams: &sync.Map{},
+	}
+}
+
+func (streamMap *TcpStreamMap) Store(key, value interface{}) {
+	streamMap.Streams.Store(key, value)
+}
+
+func (streamMap *TcpStreamMap) Delete(key interface{}) {
+	streamMap.Streams.Delete(key)
+}
+
+func (streamMap *TcpStreamMap) NextId() int64 {
+	streamMap.streamId++
+	return streamMap.streamId
```

Comment:
do we need any locker here to be sure this method to be "atomic"? (maybe I miss something important that prevent this call to be in a way that return the same id to 2 different calls)

Reply:
I didn't add that function, I just moved it from [here](https://github.com/up9inc/mizu/pull/1026/files#diff-2ed99ead342ab539b2997815d8fe2f3a1b1f5a0bf52997a1d1ae9609fe772141L34-L37) to there. It's only called in [here](https://github.com/up9inc/mizu/blob/fix/better-close-other-dissectors/tap/tcp_stream_factory.go#L73) and those calls are synchronous.